Remove duplicate test

gfontenot opened the pull request:
The null data test and the empty data test were identical. In addition, the
custom `parse` function in our empty request was duplicated, since we're
providing it for free with an extension on `Request`.

gfontenot committed:
Remove duplicate test

The null data test and the empty data test were identical. In addition,
the custom `parse` function in our empty request was duplicated, since
we're providing it for free with an extension on `Request`.

sidraval commented:
I think @paulyoung had knowingly duplicated this, but I can't remember why?

paulyoung commented:
This sort of explains it: https://github.com/thoughtbot/Swish/pull/32#discussion_r42902704

gfontenot commented:
Are we losing coverage by removing this then? I looked through these and didn't think we were, but maybe I was wrong?

tonyd256 commented:
It appears to me that the spec needs to stay but the 2 request fakes could be consolidated. Since `nil` is different than `NSData()`, the 2 fake request performers need to be used in 2 separate specs

gfontenot commented:
:+1: that makes sense

gfontenot committed:
Trim whitespace

gfontenot committed:
Clean up fake request performers

 - Introduce a ResponseData enum, that is essentially a specialized
   `Either<NSData?, [String: AnyObject]>`. This helps to consolidate the
   API and remove the need for multiple specialized request performers
 - Clean up the internals a bit

gfontenot committed:
Re-add test for if the data is `.None`

This is subtly different than empty data and worth keeping around. The
changes to the fake request performers make the differences here more
explicit.

gfontenot commented:
Re-tweaked this to keep the test around but removed the need for duplicated requests and request performers.

@sidraval @tonyd256 @paulyoung want to take another look?

jakecraige commented:
I like how this API with the `enum` turned out. :+1:

tonyd256 commented on Tests/Fakes/FakeRequestPerformer.swift:
```diff
@@ -2,55 +2,45 @@ import Foundation
-      try? NSJSONSerialization.dataWithJSONObject($0, options: NSJSONWritingOptions(rawValue: 0))
-    }
+enum ResponseData {
+  case Data(NSData?)
```
instead of having an Optional NSData case what about an `Empty` case? then remove the optionality from NSData

gfontenot replied:
I don't think that matters, since we're just passing this through. The rest of the API deals with optional `NSData`, so it feels nice (to me) to have this be explicit.

tonyd256 commented:
One comment about separating the optional data into another case ... otherwise :+1:
